### PR TITLE
add actions by bucket index

### LINKED ISSUE #1670: support bucket history in iotexscan
New API is
```
query{
  action {
    byBucketIndex(bucketIndex: 2386) {
      count
      actions{
        actHash
      }
    }
  }
}
```

http path: https://www.iotexscan.io/bucket/2345

let's reuse the same layout as https://www.iotexscan.io/action

### PR DESCRIPTION
**Pull request by hunglmtb**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind backend-change
> /kind ui-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #1670

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by hunglmtb** [comment]
pls set `ANALYTICS_API_GATEWAY_URL` in Config Var in heroku before check
`ANALYTICS_API_GATEWAY_URL=https://iotex-analytics-staging.herokuapp.com/query`

image

File: src/shared/pages/action-list-page.tsx [hunk 1]
```diff
@@ -329,14 +330,21 @@ const getActionByTypeColumns = (): Array<ColumnProps<IActionByTypeInfo>> => [
     render: text => text
   }
 ];
-export const ActionTableByType: FC<{ actionType: string }> = ({
-  actionType
-}) => {
+export const ActionTableByType: FC<{
```
Comment (CoderZhi) [inline src/shared/pages/action-list-page.tsx:333]: rename to ActionTable

Comment (CoderZhi) [reply src/shared/pages/action-list-page.tsx:333]: Mixing queries like this is very complicated and hard to follow. I would recommend splitting this component into two parts:
1. Data fetching
2. Table rendering, "<ActionTable>" accepts two parameters: actions and total count.

**Review by CoderZhi — APPROVED** [review]

File: src/shared/pages/action-list-page.tsx (hunk 1 above)
Comment (hunglmtb) [reply src/shared/pages/action-list-page.tsx]: just rename to `AnalyticActionTable` due to `ActionTable` already exists which distinguishes with action table from iotexscan.
Table rendering uses at least 4 parameters from data fetching: actions, count, state (error, loading), pagination (). It makes moving to separate parts be more TODO here
